app/models/queries/operators/custom_fields/hierarchies/equals_with_descendants.rb

```diff
@@ -0,0 +1,50 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Queries
+  module Operators
+    module CustomFields
+      module Hierarchies
+        class EqualsWithDescendants < Base
+          label "equals_with_descendants"
+          set_symbol "eq_with_descendants"
+
+          def self.sql_for_field(values, db_table, db_field)
+            items = CustomField::Hierarchy::Item.find(values)
+
+            actual_values = items.map do |item|
+              item.self_and_descendants.collect(&:id)
+            end.flatten.uniq
+
+            Equals.sql_for_field(actual_values, db_table, db_field)
+          end
```

Comment:
We can refactor this to use the HierarchyService, but this would be the thought process behind the filter.

Collect the ids, and perform a Equals filter.

Happy to hear other ideas

Reply:
Already refactored~